client/state/themes/reducer.js

```diff
@@ -346,6 +349,48 @@ export const lastQuery = createReducer( {}, {
 	} )
 } );
 
+/**
+ * Returns the updated previewing theme state
+ * The state holds information that can be used to show its preview.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export const themePreviewInfo = createReducer( {}, {
+	[ THEME_PREVIEW_DATA ]: ( state, { themeId, source } ) => ( {
+		themeId,
+		source
+	} )
+} );
+
+/**
+ * Returns the updated previewing theme state
+ * The state holds information about primary and secondary theme actions usable in preview.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export const themePreviewOptions = createReducer( {}, {
+	[ THEME_PREVIEW_OPTIONS ]: ( state, { primary, secondary } ) => ( {
+		primary,
+		secondary
+	} )
+} );
+
+/**
+ * Returns the updated previewing theme state
+ * The state reflects if Theme Preview component should be visible or not.
+ *
+ * @param  {Bool}   state  Current state
+ * @param  {Object} action Action payload
+ * @return {Bool}          Updated state
+ */
+export const isThemePreviewVisible = createReducer( false, {
```

Comment:
Oops, I meant for the selector to be called `isThemePreviewVisible` since that's a predicate 😄 
For a reducer, I'm suggesting nouns, so `themePreviewVisibility` in this case. 😉